class1/exercises.livemd

```diff
@@ -12,8 +12,30 @@ Tip: see [How to use div/2 and/or rem/2?](https://elixir-lang.org/getting-starte
 
 ```elixir
 defmodule FizzBuzz do
+  def fizzbuzz(0), do: []
+
   def fizzbuzz(n) do
-    []
+    output = ""
```

Comment:
Could you share ideas on how this exercise be completed without usage of this temporary variable?
TIPS:
Enum.map/2, anonymous/public functions with multiple clauses and guards, 

While I agree that this is correct approach, when writing Elixir code we try to write functions that are doing one of two things:
- control flow of data (e.g. by usage of pipe operator, with statement etc)
- do one thing
we can leverage this concepts to e.g split logic of value determination into multiple functions

Reply:
Changed the implementation to not use temporary variable - previous comment with "when ... ,do: ..." helped me a lot.
From the beginning I wanted to make an extendible version of FizzBuzz (to easily add new keywords).
I have made a split of the previous implementation into two modules: FizzBuzz and FizzBuzzExtendible.
The first one have a simple, easy to understand implementation. The second one is easier to maintain and/or modify.